configure.ac

```diff
@@ -1418,6 +1418,34 @@ if test "x$backtrace_method" = "x" -a "x$enable_prof_libunwind" = "x1" ; then
   fi
 fi
 
+if test `uname -s` = "Linux"
+then
+  AC_ARG_ENABLE([prof-frameptr],
+    [AS_HELP_STRING([--enable-prof-frameptr], [Use optimized frame pointer unwinder for backtracing (Linux only)])],
+  [if test "x$enable_prof_frameptr" = "xno" ; then
+    enable_prof_frameptr="0"
+  else
+    enable_prof_frameptr="1"
+    if test "x$enable_prof" = "x0" ; then
+      AC_MSG_ERROR([--enable-prof-frameptr should only be used with --enable-prof])
+    fi
+  fi
+  ],
+  [enable_prof_frameptr="0"]
+  )
+  if test "x$backtrace_method" = "x" -a "x$enable_prof_frameptr" = "x1" \
+      -a "x$GCC" = "xyes" ; then
+    JE_CFLAGS_ADD([-fno-omit-frame-pointer])
+    JE_CFLAGS_ADD([-mno-omit-leaf-frame-pointer])
```

Comment:
FYI it looks the flag might not be present on some targets, e.g. on ARM as mentioned here: https://github.com/Jguer/yay/issues/2411#issuecomment-2028471940

Since it's opt in so probably fine as is.

Question: is it true that, if we don't include these, we will not get any meaningful result, because jemalloc functions are always the top of stack, and without valid frame pointers we cannot walk through to application frames?

Reply:
From what I've seen lack of `-mno-omit-leaf-frame-pointer` doesn't break frame pointer based unwinding entirely (as long as `-fno-omit-frame-pointer` is still used - that's the critical one).

Without `-mno-omit-leaf-frame-pointer` frame pointer based unwinding can result in confusing stacks for leaf functions because a new frame isn't added by the compiler for leaf function calls. I'd say that's maybe not a big issue here as it would manifest only for jemalloc internals.

Reply:
I see. In that case, do you think we should skip `-mno-omit-leaf-frame-pointer`, for cases where it's not supported (assuming `-fno-omit-frame-pointer` is more widely available)? Like you mentioned the last frame is always jemalloc internal and we even try to filter them out: https://github.com/jemalloc/jemalloc/blob/dev/bin/jeprof.in#L2956

Reply:
It's probably easiest to not use it anywhere. I'll remove the flag.